src/world/partner/kooper.c

```diff
@@ -93,17 +151,543 @@ s32 func_802BD260_31B280(Evt* evt, s32 arg1) {
             }
             break;
     }
-    return 0;
+    return ApiStatus_BLOCK;
+}
+
+void func_802BD5F4_31B614(Npc* npc) {
+    if (D_8010C954 != NULL) {
+        D_8010C954 = NULL;
+        npc->flags = D_802BEB60_31CB80->unk_08;
+        D_802BEB60_31CB80->unk_04 = 0;
+        partner_clear_player_tracking(npc);
+    }
 }
 
-INCLUDE_ASM(s32, "world/partner/kooper", func_802BD5F4_31B614);
+#ifdef NON_EQUIVALENT
+ApiStatus func_802BD638_31B658(Evt* script, s32 isInitialCall) {
+    Camera* cam;
+    ItemEntity* itemGrabbed;
+    EncounterStatus* currentEncounter = &gCurrentEncounter;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Npc* npc = script->owner2.npc;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    CollisionStatus* collisionStatus = &gCollisionStatus;
+    f32 sp20, sp24, sp28, sp2C;
+    s32 phi_v0_4 = 0;
+    f32 temp_f20_2;
+    f32 temp_f20_3;
+    f32 temp_f20_4;
+    f32 temp_f20_5;
+    f32 temp_f20_6;
+    f32 phi_f0;
+    f32 phi_f0_3;
+    s32 tempVar;
+    f32 clamp;
+    f32 colheight;
+    
+    if (currentEncounter->unk_08 == 0) {
+        if (isInitialCall) {
+            func_802BD5F4_31B614(npc);
+            if (playerStatus->animFlags & 0x100000) {
+                return ApiStatus_DONE2;
+            }
+
+            if (D_802BEC54 == 0) {
+                tempVar = playerStatus->actionState;
+                if (tempVar < 3U) {
+                    script->functionTemp[0] = 20;
+                } else {
+                    return ApiStatus_DONE2;
+                }
+                
+            }
+
+            else if (partnerActionStatus->actionState.b[0] == 0) {
+                partnerActionStatus->actionState.b[0] = 1;
+                partnerActionStatus->actionState.b[3] = 2;
+                script->functionTemp[0] = 5;
+                npc->currentAnim.w = 0x20009;
+                D_802BEC50 = 30;
+            }
+        }
+        
+        switch (script->functionTemp[0]) {
+        case 20:
+            if (playerStatus->statusMenuCounterinputEnabledCounter == 0) {
+                if (playerStatus->decorationList == 0) {
```

Comment:
NULL - I think this is a pointer, but I could be wrong

Reply:
checked some references to playerStatus->decorationList, seems to just be an s16